front/src/modules/rollingStock/components/RollingStockEditor/CurveSpreadsheet.tsx

```diff
@@ -30,45 +35,43 @@ const CurveSpreadsheet = ({
   isDefaultCurve,
 }: CurveSpreadsheetProps) => {
   const { t } = useTranslation('rollingstock');
+  const columns = [
+    { ...keyColumn('speed', intColumn), title: t('speed') },
+    { ...keyColumn('effort', floatColumn), title: t('effort') },
+  ];
 
-  const spreadsheetCurve = useMemo(() => {
-    const { speeds, max_efforts } = selectedCurve.curve;
-    const filledMatrix: (
-      | {
-          value: string;
-        }
-      | undefined
-    )[][] =
-      speeds && max_efforts
-        ? max_efforts.map((effort, index) => [
-            {
-              value:
-                speeds[index] !== undefined ? Math.round(msToKmh(speeds[index]!)).toString() : '',
-            },
-            // Effort needs to be displayed in kN
-            { value: effort !== undefined ? Math.round(effort / 1000).toString() : '' },
-          ])
-        : [];
-    const numberOfRows = filledMatrix.length < 8 ? 8 - filledMatrix.length : 1;
-    return filledMatrix.concat(createEmptyMatrix<CellBase<string>>(numberOfRows, 2));
-  }, [selectedCurve]);
+  const [needsSort, setNeedsSort] = useState<boolean>(false);
+
+  const handleBlur = () => {
+    setNeedsSort(true);
+  };
 
-  const updateRollingStockCurve = (e: Matrix<{ value: string }>) => {
+  const updateRollingStockCurve = (newCurve: DataSheetCurve[]) => {
     if (!selectedTractionMode || !effortCurves) return;
```

Comment:
Change the type in the props of selectedTractionMode and effortCurves to make them non nullable (you can add a `&& effortCurves` in RollingStockEditorCurves l215 to check effortCurves are not null).

Then remove this test